### ISSUE TITLE
Image deconvolution example - fix to make the example code scientifically sensible

### ISSUE THREAD

**Issue by jmtayloruk** [scikit-image/scikit-image#7564]
### Description:

The example at https://scikit-image.org/docs/dev/auto_examples/filters/plot_deconvolution.html uses a strange and non-physical noise model. It _adds_ noise that has Poisson statistics with a mean of 25. The physically-correct noise model (which is what the Richardson-Lucy expects) would be for each pixel to be subject to Poisson noise with a mean equal to the noise-free value of that pixel.

For instance, a possible modification to the example code would be to change the "Add noise to image" lines to read
```python
    # Introduce Poisson noise for a scenario where the peak intensity
    # in the image is max_photon_count
    max_photon_count = 100
    astro_noisy = rng.poisson(astro * max_photon_count) / max_photon_count
    # Normalise the noisy image for skimage
    astro_noisy /= np.max(astro_noisy)
```

I'm afraid I'm not sure what the custom is in this project - do you want me to put together a pull request for this change, or does this bug report suffice?

**Comment by stefanv** [comment]
Thanks so much; this kind of report is invaluable. We could fix it, but we would also welcome your participation in the project, if you are up for it. The change will likely happen sooner if we only had to review and merge a proposal, instead of putting it together.

**Comment by mkcor** [comment]
Yes, @jmtayloruk, please go ahead and create a PR with your proposed change! Feel free to improve this gallery example overall (e.g., first sentence should read either "using the Richardson-Lucy algorithm" or "using Richardson-Lucy deconvolution"). :heart:

**Comment by mkcor** [comment]
> The example at https://scikit-image.org/docs/dev/auto_examples/filters/plot_deconvolution.html uses a strange and non-physical noise model. It adds noise that has Poisson statistics with a mean of 25.

My understanding is that perhaps this factor of 25 would cancel out (to unity) because the PSF was defined by:
```py
psf = np.ones((5, 5)) / 25
```

But I can't understand the division by 255, here:
https://github.com/scikit-image/scikit-image/blob/efe339b09ee7d9d8eb163d6750005ab0bef703b8/doc/examples/filters/plot_deconvolution.py#L35

Because pixel values have been ~1 ever since the astronaut image was converted to grayscale.

**Comment by jmtayloruk** [comment]
> > The example at https://scikit-image.org/docs/dev/auto_examples/filters/plot_deconvolution.html uses a strange and non-physical noise model. It adds noise that has Poisson statistics with a mean of 25.
> 
> My understanding is that perhaps this factor of 25 would cancel out (to unity) because the PSF was defined by:
> 
> ```python
> psf = np.ones((5, 5)) / 25
> ```
> 
> But I can't understand the division by 255, here:
> 
> https://github.com/scikit-image/scikit-image/blob/efe339b09ee7d9d8eb163d6750005ab0bef703b8/doc/examples/filters/plot_deconvolution.py#L35
> 
> Because pixel values have been ~1 ever since the astronaut image was converted to grayscale.

I wouldn't want to second-guess the thinking of the original code author, but with all due respect to them I don't think it's worth trying too hard to figure out their reasoning, because the constant value for `lam` was non-physical and inconsistent with the expectations of the Richardson-Lucy algorithm. Fix is working its way to final approval in #7589, as you know.
